python/tvm/meta_schedule/testing/distributed_measure_candidates.py

```diff
@@ -0,0 +1,199 @@
+    tuning_records = database.get_all_tuning_records()
+    for record in tuning_records:
+        candidates.append(context.get_measure_candidate(record.workload.mod, record.trace))
+    for idx in range(0, len(candidates), args.batch_size):
+        batch_candidates = candidates[idx : idx + args.batch_size]
+        context.set_measure_candidates(batch_candidates)
+        build_start_time = time.time()
+        context.send_to_builder(builder)
+        build_end_time = time.time()
+        context.send_to_runner(runner)
+        batch_runner_results = context.join()
+        run_end_time = time.time()
+        runner_results.extend(batch_runner_results)
+        for i, result in enumerate(context.builder_results):
+            if result.error_msg is None:
+                ms.utils.remove_build_dir(result.artifact_path)
+            else:
+                build_fail_indices.append(i + idx)
+        context.clear_measure_state()
+        build_time += build_end_time - build_start_time
+        run_time += run_end_time - build_end_time
+
+    model_name, workload_name = database.path_workload.split("/")[-2:]
+    record_name = database.path_tuning_record.split("/")[-1]
+    new_database = ms.database.JSONDatabase(
+        path_workload=os.path.join(args.result_cache_dir, model_name, workload_name),
+        path_tuning_record=os.path.join(args.result_cache_dir, model_name, record_name),
+    )
+    workload = tuning_records[0].workload
+    new_database.commit_workload(workload.mod)
+    for i, (record, result) in enumerate(zip(tuning_records, runner_results)):
+        if result.error_msg is None:
+            new_database.commit_tuning_record(
+                ms.database.TuningRecord(
+                    trace=record.trace,
+                    workload=workload,
+                    run_secs=[v.value for v in result.run_secs],
+                    target=Target(args.target),
+                )
+            )
+        else:
+            run_fail_indices.append(i)
+    fail_indices_name = workload_name[:-13] + "failed_indices.txt"
+    with open(
+        os.path.join(args.result_cache_dir, model_name, fail_indices_name), "w", encoding="utf8"
+    ) as file:
+        file.write(" ".join([str(n) for n in run_fail_indices]))
+    print(
+        f"Builder time: {build_time}, Runner time: {run_time}\n\
+            Failed number of builds: {len(build_fail_indices)},\
+            Failed number of runs: {len(run_fail_indices)}"
+    )
+
+
+args = _parse_args()  # pylint: disable=invalid-name
+
+
+def main():
+    builder = ms.builder.LocalBuilder(timeout_sec=args.builder_timeout_sec)
+    runner = ms.runner.RPCRunner(
+        rpc_config=ms.runner.RPCConfig(
+            tracker_host=args.rpc_host,
+            tracker_port=args.rpc_port,
+            tracker_key=args.rpc_key,
+            session_timeout_sec=args.runner_timeout_sec,
+        ),
+        evaluator_config=ms.runner.EvaluatorConfig(
+            number=1,
```

Comment:
Usually we use `number=3`

Reply:
I think we might want to measure only once though. What do you think?